### PR TITLE
work over the intro

### PR DESCRIPTION
**Pull request by konnov**
This PR extends the intro. Also, I've moved the discussion right into the intro. Otherwise, the reader would not find our discussions after 30+ pages.

### CODE REVIEW

File: reports/intro.tex [hunk 1]
```diff
@@ -6,11 +6,24 @@ \section{Introduction}
...
+was to demonstrate accountable safety of this protocol by the means of model
+checking.
+
+We have chosen the specification language~\tlap{} and the model checker
+Apalache for the following reasons.\ \tlap{} remains a goto language for
+specifying consensus algorithms. Among the rich spectrum of
+specifications~\cite{tla-examples}, the most notable for our project are the
+specifications of Paxos~\cite{lamport2001paxos}, Raft~\cite{Ongaro14}, and
+Tendermint~\cite{abs-1807-04938,TendermintSpec2020}. Since consensus algorithms
+are quite challenging for the classical model checkers such as TLC, we chose
```
Comment (thpani) [inline reports/intro.tex:21]:
```suggestion
are quite challenging for classical model checkers like TLC, we choose
```

File: reports/intro.tex [hunk 2]
```diff
@@ -6,11 +6,24 @@ \section{Introduction}
...
+Apalache~\cite{Apalache2024,KT19,KonnovKM22}. Apalache was used for model
```
Comment (thpani) [inline reports/intro.tex:22]:
```suggestion
Apalache~\cite{Apalache2024,KT19,KonnovKM22}. Apalache has been used for model
```

File: reports/intro.tex [hunk 3]
```diff
@@ -35,50 +48,68 @@ \section{Introduction}
...
+        specification (such as the Python code). This specification is
+        discussed in Section~\ref{sec:spec3}.
+
+    \item \SpecFour{}: This is an extension of~\SpecThree{} that contains
+        an inductive invariant in~\texttt{spec4/ffg\_inductive.tla}.
+        This specification is discussed in Section~\ref{sec:spec4}.
+
+    \item \SpecFourB{} contains further abstractions and decomposition of
+        configurations. This is the first~\tlap{} specification that allowed us
+        to show accountable safety for models of very small sizes. This
```
Comment (thpani) [inline reports/intro.tex:95]:
```suggestion
        to show accountable safety for models of very small size. This
```

File: reports/intro.tex [hunk 4]
```diff
@@ -35,50 +48,68 @@ \section{Introduction}
...
+        specification is discussed in Section~\ref{sec:spec4b}.
+
+    \item \SpecThreeB{} contains a specification in SMT using the theory of
+        finite sets and cardinalities. In combination with the solver
+        CVC5~\cite{BarbosaBBKLMMMN22}, this specification allowed us to push
```
Comment (thpani) [inline reports/intro.tex:100]:
```suggestion
        CVC5~\cite{BarbosaBBKLMMMN22}, this specification allows us to push
```

File: reports/intro.tex [hunk 5]
```diff
@@ -35,50 +48,68 @@ \section{Introduction}
...
+        verification of accountable safety even further. This specification is
+        discussed in Section~\ref{sec:smt}.
+
+    \item \SpecThreeC{} contains a specification in
+        Alloy~\cite{jackson2012software,alloytools}. With this specification,
+        we managed to check all small configurations that cover the base case
```
Comment (thpani) [inline reports/intro.tex:106]:
```suggestion
        we manage to check all small configurations that cover the base case
```

File: reports/spec2.tex [hunk 6]
```diff
@@ -132,13 +132,29 @@ \subsection{Checking the Specification}
...
+
+To further evaluate \SpecTwo{}, we fix the block graph --- this way the solver
+only has to reason about voting. We encode three example block graphs: a
+single, linear chain (\texttt{SingleChain}), a minimal forked chain of three
+blocks (\texttt{ShortFork}), and a forest of disconnected chains
 (\texttt{Forest}). Table~\ref{tab:spec2_fixed} shows the results of model
 checking \SpecTwo{} for these fixed block graphs.
 
+To understand the structure of those small graphs, see
+Figure~\ref{fig:block-graphs} shows small graphs. In the simple
```
Comment (thpani) [inline reports/spec2.tex:146]: There's something wrong with this sentence

**Review by thpani — APPROVED** [review]
LGTM, I've added some suggestions. 

In general, I would stay in present tense as much as possible.

File: reports/intro.tex [hunk 7]
```diff
@@ -6,11 +6,24 @@ \section{Introduction}
...
 \paragraph{}\colorbox{yellow}{TODO:}
 
-\begin{itemize}
-  \item what we want to model-check
-  \item why \tlap{}
-  \item introduce Apalache
-\end{itemize}
+We have started this project with the Python specification of accountable
+safety in the 3SF Protocol\footnote{URL to the Python specification:
+\url{https://github.com/freespek/ssf-mc/3sf.txt}}. The main goal of the project
```
[the remark this answers is not in the document]
Comment (konnov) [reply reports/intro.tex:11]: True, but it is only a link to the module, not to the spec itself

File: reports/intro.tex [hunk 8]
```diff
@@ -6,11 +6,24 @@ \section{Introduction}
...
+was to demonstrate accountable safety of this protocol by the means of model
+checking.
+
+We have chosen the specification language~\tlap{} and the model checker
+Apalache for the following reasons.\ \tlap{} remains a goto language for
+specifying consensus algorithms. Among the rich spectrum of
+specifications~\cite{tla-examples}, the most notable for our project are the
+specifications of Paxos~\cite{lamport2001paxos}, Raft~\cite{Ongaro14}, and
+Tendermint~\cite{abs-1807-04938,TendermintSpec2020}. Since consensus algorithms
```
[the remark this answers is not in the document]
Comment (konnov) [reply reports/intro.tex:20]: Yeah, it is still work-in-progress :)

**Pull request merged**